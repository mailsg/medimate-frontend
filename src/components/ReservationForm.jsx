import React, { useState } from 'react';

function ReservationForm() {
  const [formData, setFormData] = useState({
    appointmentDate: '',
    appointmentTime: '',
    duration: '',
    userId: '',
    doctorId: '',
  });

  const handleSubmit = async (e) => {
    e.preventDefault();

    // Send formData to your backend API to create a new appointment
    try {
      const response = await fetch('/api/appointments', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(formData),
      });

      if (response.ok) {
        // Handle successful reservation (e.g., update UI)
        // You can add code here to indicate a successful reservation.
      } else {
        // Handle reservation error (e.g., update UI with an error message)
        // You can add code here to display an error message to the user.
      }
    } catch (error) {
      // Handle unexpected errors (e.g., update UI with a generic error message)
      // You can add code here to display a generic error message to the user.
    }
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData({ ...formData, [name]: value });
  };

  return (
    <div className="reservation-container">
      <h2>Reservation Form</h2>
      <form className="reservation-form" onSubmit={handleSubmit}>
        <div className="form-group">
          <label htmlFor="appointmentDate">Date:</label>
          <input
            type="date"
            id="appointmentDate"
            name="appointmentDate"
            value={formData.appointmentDate}
            onChange={handleChange}
            required
          />
        </div>
        <div className="form-group">
          <label htmlFor="appointmentTime">Time:</label>
          <input
            type="time"
            id="appointmentTime"
            name="appointmentTime"
            value={formData.appointmentTime}
            onChange={handleChange}
            required
          />
        </div>
        <div className="form-group">
          <label htmlFor="duration">Duration (in minutes):</label>
          <input
            type="number"
            id="duration"
            name="duration"
            value={formData.duration}
            onChange={handleChange}
            required
          />
        </div>
        <button type="submit" className="submit-button">
          Make Reservation
        </button>
      </form>
    </div>
  );
}

export default ReservationForm;
