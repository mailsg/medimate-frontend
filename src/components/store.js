import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './redux/doctorSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
  },
});

export default store;
