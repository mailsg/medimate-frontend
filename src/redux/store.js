import { configureStore } from '@reduxjs/toolkit';
import AppointmentSlice from './appointmentSlice';
import DoctorSlice from './doctorSlice';
import appReducer from './adddoctorSlice';

const store = configureStore({
  reducer: {
    doctor: DoctorSlice,
    appointment: AppointmentSlice,
    app: appReducer,
  },
});

export default store;
