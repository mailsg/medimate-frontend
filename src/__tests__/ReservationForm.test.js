import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import ReservationForm from '../components/ReservationForm';
import store from '../redux/store';

describe('Reservation Form Component', () => {
  test('renders Reservation Form component', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ReservationForm />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
