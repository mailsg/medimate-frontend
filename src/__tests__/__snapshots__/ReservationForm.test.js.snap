// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation Form Component renders Reservation Form component 1`] = `
<div>
  <div
    class="form reservation-container"
  >
    <h1
      class="reserve-form-header"
    >
      Reservation Form
    </h1>
    <form
      class="form reservation-form"
    >
      <div
        class="form-group"
      >
        <select
          class="select-elm"
          id="doctorId"
          name="doctor_id"
          required=""
        >
          <option
            value=""
          >
            Select a Doctor
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form"
          id="appointmentDate"
          name="appointment_date"
          placeholder="Date of Appointment"
          required=""
          type="date"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form"
          id="appointmentTime"
          name="appointment_time"
          placeholder="Time of Appointment"
          required=""
          type="time"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form"
          id="duration"
          name="duration"
          placeholder="Duration (in minutes)"
          required=""
          type="number"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form"
          id="location"
          name="location"
          placeholder="Location"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="btn-container"
      >
        <button
          class="submit-button"
          type="submit"
        >
          Make Reservation
        </button>
      </div>
    </form>
  </div>
</div>
`;
